### PR TITLE
bpo-28326: Fixes multiprocessing.Process depends on sys.stdout being open

### PR DESCRIPTION
**Pull request by tiagoantao**
This fixes bug 28326, please see https://bugs.python.org/issue28326 for more details

### CODE REVIEW

**Comment by pitrou** [comment]
Sorry for not noticing this PR before. Will take a look right now.

File: Lib/test/test_multiprocessing_main_handling.py [hunk 1]
```diff
@@ -65,6 +65,13 @@ def f(x):
             raise RuntimeError("Timed out waiting for results")
     results.sort()
     print(start_method, "->", results)
+    sys.stdout.close()
```
Comment (pitrou) [inline Lib/test/test_multiprocessing_main_handling.py:68]: While I understand it's practical to do so, it's a bit weird to add your test there, as it doesn't really have anything to do with `__main__` handling. Instead, I would expect it to be somewhere in `_TestProcess` in `_test_multiprocessing.py`.

**Comment by pitrou** [comment]
Note you'll need to add a NEWS entry using the [blurb tool](https://github.com/python/core-workflow/tree/master/blurb).

File: Lib/multiprocessing/popen_fork.py [hunk 2]
```diff
@@ -14,8 +14,10 @@ class Popen(object):
     method = 'fork'
 
     def __init__(self, process_obj):
-        sys.stdout.flush()
-        sys.stderr.flush()
+        if not sys.stdout.closed:
```
Comment (terryjreedy) [inline Lib/multiprocessing/popen_fork.py:17]: Either change this to
`        if sys.stdout is not None and not sys.stdout.closed:`

or change the entire statement to
```
        try:
            sys.stdout.flush()
        except (AttributeError, ValueError):  # stdout None or closed.
            pass
```
and the equivalent for stderr.  I slightly prefer the latter (these exceptions should be rare), but if Antoine expresses a preference for the former, use that.

File: Lib/test/test_multiprocessing_main_handling.py (hunk 1 above)
Comment (terryjreedy) [reply Lib/test/test_multiprocessing_main_handling.py:68]: I also wondered about the placement, so I think you should follow Antoine's suggestion.

**Review by terryjreedy — CHANGES_REQUESTED** [review]
The news blurb requirement is new since this was submitted.  I added and deleted the tag to trigger it.

If you fix the code and test and add a news entry, I think you can also add yourself to Misc/ACKS.

**Comment by tiagoantao** [comment]
I have submitted this a long long time ago. I completely lost track of it and have moved on. A suggestion: when you have new people trying to help please be a little bit more responsive.

**Comment by pitrou** [comment]
In any case this was fixed in https://github.com/python/cpython/pull/4073

**Pull request closed without merging**